tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -532,20 +532,22 @@ def do_rolling_bounce(self, processor, counter, current_generation):
                         log_monitor.wait_until("Sent a version 8 subscription and got version 7 assignment back (successful version probing). Downgrade subscription metadata to commonly supported version 7 and trigger new rebalance.",
                                                timeout_sec=60,
                                                err_msg="Could not detect 'successful version probing' at upgrading node " + str(node.account))
-                    else:
-                        log_monitor.wait_until("Sent a version 8 subscription and got version 7 assignment back (successful version probing). Downgrade subscription metadata to commonly supported version 8 and trigger new rebalance.",
+                        log_monitor.wait_until("Detected that the assignor requested a rebalance. Rejoining the consumer group to trigger a new rebalance.",
```

Comment:
I know that this check was here in some fashion before, but I'm drawing a blank on why we need to verify this log line. It seems like _just_ checking the version number logs and nothing else would be the key to a long and happy life.

Reply:
I think the idea is to verify that the actual version probing rebalance takes place, ie that the partition assignor actually handles the version probing once it's detected. And that it signals to the stream thread which also handles it correctly in turn. But idk -- I've probably broken and fixed the version probing test 2 or 3 times now due to this one line in particular so I'd be happy to see it go. I probably have too much bad history to make an unbiased call here though 😄

Reply:
I guess what we really would need is a way to check if a group stabilized. We try to do that by verifying that the generations of the processors are synced. However, I ran into cases where all processor had the same generation, but one processor did not have any tasks assigned, because in that specific rebalance the corresponding partitions were revoked from the other processors. So we would actually need to check if they have the highest generation is in sync across the processors AND if all processors have at least one task assigned (AND if all tasks were assigned).

Reply:
Thanks, all. This doesn't seem like the best way to verify what we're trying to verify, but it also seems about the same as before. I'm happy to leave this here for now.

If/when the test breaks again, I'd prefer for us to put in a more reliable and direct mechanism.